### PR TITLE
[15.0][IMP] stock_request: Show error in stock.request.order in the "Confirm" or "Submit" buttons if there are no items.

### PR DESCRIPTION
**Pull request by victoralmau**
FWP from 14.0: https://github.com/OCA/stock-logistics-warehouse/pull/1511

Changes done:
- [x] Show error in `stock.request.order` in the "_Confirm_" or "_Submit_" buttons if there are no items.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT39761

### CODE REVIEW

File: stock_request/views/stock_request_order_views.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
                         string="Confirm"
                         type="object"
                         attrs="{'invisible': [('state', 'not in', ['draft'])]}"
+                        groups="stock_request.group_stock_request_manager"
```
Comment (rousseldenis) [inline stock_request/views/stock_request_order_views.xml:33]: Why changing this behaviour ?

Comment (rousseldenis) [reply stock_request/views/stock_request_order_views.xml:33]: @victoralmau Stock request users should have access to confirmations. If you want to restrict, please either do it on your side or add an option to configure this behaviour.

Comment (victoralmau) [reply stock_request/views/stock_request_order_views.xml:33]: Perhaps another possible approach would be to add a setting to define who can use confirm button (user or manager). What do you think about this?

Comment (rousseldenis) [reply stock_request/views/stock_request_order_views.xml:33]: Yes and protect the action method through that

File: stock_request/models/res_company.py [hunk 2]
```diff
@@ -12,3 +12,11 @@ class ResCompany(models.Model):
     stock_request_allow_virtual_loc = fields.Boolean(
         string="Allow Virtual locations on Stock Requests"
     )
+    stock_request_confirm_user_permission = fields.Selection(
+        selection=[
+            ("user", "User"),
+            ("manager", "Manager"),
+        ],
+        default="user",
```
Comment (pedrobaeza) [inline stock_request/models/res_company.py:20]:
```suggestion
        default="manager",
```

Comment (rousseldenis) [reply stock_request/models/res_company.py:20]: Nope as this changes the base behaviour from the beginning of this module. As already said, you can override that in your project.

Comment (pedrobaeza) [reply stock_request/models/res_company.py:20]: Yes, but that's the intended behavior. It's a nonsense to have 2 steps (submit and confirm), when both profiles can make by default both. We'll bump major version.

**Comment by rousseldenis** [comment]
@JordiBForgeFlow @kittiu I want your advice here

File: stock_request/models/res_company.py (hunk 2 above)
Comment (rousseldenis) [reply stock_request/models/res_company.py:20]: > We'll bump major version.

Yes, but it will be applied on further version. I think about migrations

Comment (pedrobaeza) [reply stock_request/models/res_company.py:20]: We can put a migration script for those coming from previous versions that preserve the other, but I really think it's a mistake to let it as it was before. If Jordi/Kitti thinks the same, we can even remove this mechanism and let the new default only.

Comment (rousseldenis) [reply stock_request/models/res_company.py:20]: @pedrobaeza I've reviewed 14.0 module and I don't find any submit state.

There are only 'draft' one (default) and open one. Could you review that ?

Comment (pedrobaeza) [reply stock_request/models/res_company.py:20]: OK, I see the problem. The new module `stock_request_submit` is the one to fix:

- It should change the button "Confirm" to launch the method to put the request into submit.
- The button submit should call the method `confirm`.
- This button is the one to be protected by manager permission.

**Comment by victoralmau** [comment]
I close this PR because finally it is not necessary to add any changes to v15 (changes are added in `stock_request_submit`).

**Comment by pedrobaeza** [comment]
But what about the message when 0 stock request lines?

**Comment by victoralmau** [comment]
> But what about the message when 0 stock request lines?

Ok sorry, can you reopen it?

**Comment by pedrobaeza** [comment]
It's nos possible anymore it seems. Do another one, and the fw-port of the submit thing is also pending, no?

**Pull request closed without merging**